RMS/BufferedCapture.py

```diff
@@ -110,6 +115,43 @@ def __init__(self, array1, startTime1, array2, startTime2, config, video_file=No
         self.frame_shape = None
         self.convert_to_gray = False
 
+        # Params to help with camera mode switching
+        self.daytime_mode = daytime_mode
+
+        if self.config.save_frames:
+
+            # Params for raw frame saving
+            self.num_raw_frames = 10
+            self.raw_frame_count = 0
+
+            # Initialize shared values for frame + timestamp saving
+            self.startRawTime1 = Value('d', 0.0)
+            self.startRawTime2 = Value('d', 0.0)
+
+
+            # Initialize shared memory arrays for raw frame saving
+            self.sharedRawArrayBase = Array(ctypes.c_uint8, self.num_raw_frames * config.height * config.width)
+            self.sharedRawArray = np.ctypeslib.as_array(self.sharedRawArrayBase.get_obj())
+            self.sharedRawArray = self.sharedRawArray.reshape(self.num_raw_frames, config.height, config.width)
+
+            self.sharedRawArrayBase2 = Array(ctypes.c_uint8, self.num_raw_frames * config.height * config.width)
+            self.sharedRawArray2 = np.ctypeslib.as_array(self.sharedRawArrayBase2.get_obj())
+            self.sharedRawArray2 = self.sharedRawArray2.reshape(self.num_raw_frames, config.height, config.width)
+

```

Comment:
Should we initialize the arrays after the frame shape is known? And handle the switch between greyscale and 3 channels?

Reply:
maybe first initialize the arrays as None:
```python
self.sharedRawArrayBase = None
self.sharedRawArray = None
self.sharedRawArrayBase2 = None
self.sharedRawArray2 = None
self.rawFrameSaver = None
```
 Then initialize with a method when the shape is known:
 ```python
def initRawFrameArrays(self, frameShape):
        """Initialize the raw frame arrays once we know the frame shape"""
        if not self.config.save_frames:
            return

        # Calculate buffer size based on frame shape
        if len(frame_shape) == 3:
            # Color frame (height, width, channels)
            bufferSize = self.num_raw_frames * frameShape[0] * frameShape[1] * frameshape[2]
            arrayShape = (self.num_raw_frames, frameShape[0], frameShape[1], frameShape[2])
        else:
            # Grayscale frame (height, width)
            bufferSize = self.num_raw_frames * frameShape[0] * frameShape[1]
            arrayShape = (self.num_raw_frames, frameShape[0], frameShape[1])

        # Initialize shared memory arrays for raw frame saving
        self.sharedRawArrayBase = Array(ctypes.c_uint8, buffer_size)
        self.sharedRawArray = np.ctypeslib.as_array(self.sharedRawArrayBase.get_obj())
        self.sharedRawArray = self.sharedRawArray.reshape(arrayShape)

        self.sharedRawArrayBase2 = Array(ctypes.c_uint8, buffer_size)
        self.sharedRawArray2 = np.ctypeslib.as_array(self.sharedRawArrayBase2.get_obj())
        self.sharedRawArray2 = self.sharedRawArray2.reshape(arrayShape)

        # Initialize the raw frame saver
        self.rawFrameSaver = RawFrameSaver(self.saved_frames_dir, 
                                           self.sharedRawArray, self.startRawTime1, 
                                           self.sharedRawArray2, self.startRawTime2, 
                                           self.sharedTimestamps, self.sharedTimestamps2,
                                           self.config)
```